### PR TITLE
AssetRegistry: protect asset_registry migration

### PR DESCRIPTION
**Pull request by lemunozm**
# Description

The check for the expected state of `asset_registry` migration is checked under `try-runtime`, if the chain state changes, the new state sets an empty vector, removing all currencies.

This PR modifies the migration only to take effect if the chain state is the expected state.

- [x] Checked manually with real altair environment

### CODE REVIEW

File: runtime/altair/src/migrations.rs [hunk 1]
```diff
@@ -62,31 +62,9 @@ mod asset_registry {
...
-			loc_count: u32,
-			meta_count: u32,
-		) -> Vec<(
-			CurrencyId,
-			orml_asset_registry::AssetMetadata<Balance, CustomMetadata>,
-		)> {
-			match (loc_count, meta_count) {
-				(loc, meta)
-					if (loc, meta) == (ALTAIR_ASSET_LOC_COUNT, ALTAIR_ASSET_METADATA_COUNT) =>
-				{
```
Comment (lemunozm) [inline runtime/altair/src/migrations.rs:80]: Check moved to the migration itself

File: runtime/common/src/migrations/asset_registry_xcmv3.rs [hunk 2]
```diff
@@ -185,7 +175,7 @@ impl<
 {
 	fn get_key_counts() -> (u32, u32) {
 		// let loc_count =
-		// orml_asset_registry::LocationToAssetId::<T>::iter_keys().count() as u32;
+		//  orml_asset_registry::LocationToAssetId::<T>::iter_keys().count() as u32;
```
Comment (lemunozm) [inline runtime/common/src/migrations/asset_registry_xcmv3.rs:178]: Not sure why the commented line doesn't work

**Comment by lemunozm** [comment]
BTW, not sure if this migration should still be there or can be removed from altair (?)

**Comment by lemunozm** [comment]
Confirmed, it's needed

**Comment by wischli** [comment]
> Confirmed, it's needed

Yes. Next release must include Altair for sure. We have not treated it properly due to limited capacity.

**Comment by lemunozm** [comment]
Checked against `altair`:
```
2023-10-17 14:01:08.694  INFO main runtime_common::migrations::asset_registry_xcmv3: 💎 AssetRegistryMultilocationToXCMV3: on_runtime_upgrade: started
2023-10-17 14:01:08.695  INFO main runtime_common::migrations::asset_registry_xcmv3: 💎 AssetRegistryMultilocationToXCMV3: Found 5 LocationToAssetId keys
2023-10-17 14:01:08.695  INFO main runtime_common::migrations::asset_registry_xcmv3: 💎 AssetRegistryMultilocationToXCMV3: Found 5 Metadata keys
2023-10-17 14:01:08.695  INFO main runtime_common::migrations::asset_registry_xcmv3: 💎 AssetRegistryMultilocationToXCMV3: Cleared all LocationToAssetId entries successfully
2023-10-17 14:01:08.695  INFO main runtime_common::migrations::asset_registry_xcmv3: 💎 AssetRegistryMultilocationToXCMV3: LocationToAssetId clearing iteration result. backend: 5 unique: 5 loops: 5
2023-10-17 14:01:08.695  INFO main runtime_common::migrations::asset_registry_xcmv3: Cleared all Metadata entries successfully
2023-10-17 14:01:08.695  INFO main runtime_common::migrations::asset_registry_xcmv3: 💎 AssetRegistryMultilocationToXCMV3: Metadata clearing iteration result. backend: 5 unique: 5 loops: 5
2023-10-17 14:01:08.695  INFO main runtime_common::migrations::asset_registry_xcmv3: 💎 AssetRegistryMultilocationToXCMV3: Starting migration of 4 assets
2023-10-17 14:01:08.695 DEBUG main runtime_common::migrations::asset_registry_xcmv3: Migrating asset: Native
2023-10-17 14:01:08.695 DEBUG main runtime_common::migrations::asset_registry_xcmv3: Migrating asset: ForeignAsset(1)
2023-10-17 14:01:08.695 DEBUG main runtime_common::migrations::asset_registry_xcmv3: Migrating asset: ForeignAsset(2)
2023-10-17 14:01:08.696 DEBUG main runtime_common::migrations::asset_registry_xcmv3: Migrating asset: ForeignAsset(3)
2023-10-17 14:01:08.696  INFO main runtime_common::migrations::asset_registry_xcmv3: 💎 AssetRegistryMultilocationToXCMV3: on_runtime_upgrade: completed!
```

**Review by wischli — APPROVED** [review]

File: runtime/common/src/migrations/asset_registry_xcmv3.rs (hunk 2 above)
Comment (lemunozm) [reply runtime/common/src/migrations/asset_registry_xcmv3.rs:178]: > Keys which cannot be decoded are not counted whereas count_storage_keys does not try to decode and thus counts correctly.

Good point! Thanks for expanding it.

**Comment by lemunozm** [comment]
Thanks for your fast review as always @William!

**Pull request merged**